flycheck.el

```diff
@@ -6248,6 +6249,66 @@ Requires DMD 2.066 or newer.  See URL `http://dlang.org/'."
          (one-or-more " ") (message) line-end))
   :modes d-mode)
 
+(defun flycheck-elixir--find-default-directory (checker)
+  "Come up with a suitable default directory to run CHECKER in.
+
+This will either be the directory that contains `mix.exs' or,
+if no such file is found in the directory hierarchy, ths directory
+of the current file."
+  (or
+   (locate-dominating-file (buffer-file-name) "mix.exs")
+   (file-name-directory (buffer-file-name))))
+
+(defun flycheck-elixir--parse-dogma-json (output checker buffer)
+  "Parse dogma errors from JSON OUTPUT.
+
+CHECKER and BUFFER denoted the CHECKER that returned OUTPUT and
+the BUFFER that was checked respectively.
+
+See URL `https://github.com/lpil/dogma' for more information
+about dogma."
+  (let* ((json-object-type 'alist)
+         (json-array-type  'list)
+         (dogma-json-output
+          (car (cdr (assq 'files (json-read-from-string output)))))
+         (dogma-errors-list (cdr (assq 'errors dogma-json-output)))
+         (dogma-filename (cdr (assq 'path dogma-json-output)))
+         errors)
+    (dolist (emessage dogma-errors-list)
+      (let-alist emessage
+        (push (flycheck-error-new-at
+               .line
+               1
+               'error .message
+               :id .rule
+               :checker checker
+               :buffer buffer
+               :filename dogma-filename)
+              errors)))
+    (nreverse errors)))
+
+(defun flycheck-elixir--check-for-dogma ()
+  "Check if the `dogma' subcommand is available for
+from mix by parsing the output of `mix help'.
+
+Used as a predicate for enabling the checker."
+  (let ((mix (flycheck-checker-executable 'elixir-dogma))
+        (found))
+    (dolist (line (process-lines mix "help" ))
+      (if (string-match "dogma" line)
```

Comment:
Use `string-match-p` instead of `string-match` here.  We don't need to modify the match data as we don't use it anywhere.